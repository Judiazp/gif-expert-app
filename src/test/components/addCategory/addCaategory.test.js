import React from "react";
import '@testing-library/jest-dom'
import { shallow } from "enzyme";
import AddCategory from "../../../components/addCategory";

describe('Pruebas en <AddCategory />', () => {
    const setCategories = jest.fn()
    let wrapper = shallow( <AddCategory setCategories={ setCategories } />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } />);

    })

    test('debe mostrar el componente AddCategory', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Un string'
        input.simulate('change', { target: { value }});
    });

    test('No debe postear la informacion con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){}})
        expect( setCategories ).not.toHaveBeenCalled()
    });

    test('debe llamar setCategories y limpiar la caja de texto ', () => {
       
        const value = 'Un string'
        wrapper.find('input').simulate('change', { target: { value }})
        wrapper.find('form').simulate('submit', { preventDefault(){}})

        expect( setCategories ).toHaveBeenCalled()
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function))
        expect( wrapper.find('input').prop('value')).toBe('')

    });
})