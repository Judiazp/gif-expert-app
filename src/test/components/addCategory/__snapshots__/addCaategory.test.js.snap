// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrar el componente AddCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar Gifs"
    type="text"
    value=""
  />
</form>
`;
